modules/server_new/src/main/scala/observe/server/ObserveEngine.scala

```diff
@@ -1706,8 +1706,10 @@ object ObserveEngine {
         // The sequence could be empty
         case Nil => Nil
         // Find first Pending ObserveStep when no ObserveStep is Running and mark it as Running
+        // When waiting user prompt, the sequence is Running, but no steps are.
         case steps
-            if Sequence.State.isRunning(st) && steps.forall(_.status =!= StepState.Running) =>
+            if (Sequence.State.isRunning(st) && !Sequence.State.isWaitingUserPrompt(st)) && steps
```

Comment:
I think this is the key-change. so when waiting no steps are running and it should be equivalent to the previous contidiotn

Reply:
Correct